app/src/main/java/com/example/plantsapp/presentation/ui/notification/broadcastreceiver/NotificationBroadcastReceiver.kt

```diff
@@ -0,0 +1,50 @@
+package com.example.plantsapp.presentation.ui.notification.broadcastreceiver
+
+import android.app.NotificationManager
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import androidx.annotation.CallSuper
+import com.example.plantsapp.domain.model.TaskKeys
+import com.example.plantsapp.domain.usecase.CompleteTaskUseCase
+import com.example.plantsapp.presentation.ui.notification.utils.notificationId
+import com.example.plantsapp.presentation.ui.notification.utils.taskFrequency
+import com.example.plantsapp.presentation.ui.notification.utils.taskId
+import com.example.plantsapp.presentation.ui.notification.utils.taskKey
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.runBlocking
+import java.util.Date
+import javax.inject.Inject
+
+abstract class HiltBroadcastReceiver : BroadcastReceiver() {
+    @CallSuper
+    override fun onReceive(context: Context, intent: Intent?) = Unit
+}
+
+@AndroidEntryPoint
+class NotificationBroadcastReceiver : HiltBroadcastReceiver() {
+
+    @Inject
+    lateinit var completeTaskUseCase: CompleteTaskUseCase
+
+    override fun onReceive(context: Context, intent: Intent?) {
+        super.onReceive(context, intent)
+        intent?.let {
+            val task = TaskKeys.getFromKey(it.taskKey)
+                .toTask(
+                    taskId = it.taskId,
+                    frequency = it.taskFrequency
+                )
+
+            runBlocking {
```

Comment:
Broadcast Receiver runs on UI thread, so it is not best idea to block it

Also read this part (as on screenshot)
https://developer.android.com/guide/components/broadcasts#effects-process-state
![image](https://user-images.githubusercontent.com/23121846/143874224-1f6a47dd-0944-4bf3-adfc-6aaf0af558ac.png)